apps/classroomio-com/src/routes/tools/progress/components/store.ts

```diff
@@ -0,0 +1,118 @@
+import { writable } from 'svelte/store';
+import type { Writable } from 'svelte/store';
+
+export interface Mood {
+  text: string;
+  iconSrc: string;
+}
+export interface Source {
+  src: string;
+}
+
+interface Modal {
+  open: boolean;
+}
+
+export interface HtmlBody {
+  name: string;
+  learning: string;
+  progress: number;
+  avatar: string;
+  background: string;
+  mood: {
+    text: string;
+    iconSrc: string;
+  };
+}
+
+// report html
+export const htmlBody: Writable<HtmlBody> = writable({
+  name: '',
+  learning: 'Tell us what you are learning',
+  progress: 10,
+  avatar: '',
+  background: '',
+  mood: {
+    text: '',
+    iconSrc: ''
+  }
+});
+
+// store for the node binded to the report component for the component to image conversion
+export const nodeStore = writable(null);
+
+export const openMood: Writable<Modal> = writable({
+  open: false
+});
+
+export const openAvatar: Writable<Modal> = writable({
+  open: false
+});
+
+export const openBackground: Writable<Modal> = writable({
+  open: false
+});
+
+export const openFullscreen: Writable<Modal> = writable({
+  open: false
+});
```

Comment:
There is too much redundancy here, for this particular store we don't need individual stores for just a boolean.

Refactor this to:

```ts
interface OpenModal {
  avatar: boolean;
  background: boolean;
  fullscreen: boolean;
  mood: boolean;
}
export const openModal: Writable<OpenModal> = writable({
  avatar: false,
  background: false,
  fullscreen: false,
  mood: false,
});
```